hoomd/util.py

```diff
@@ -24,6 +25,44 @@ def _bad_iterable_type(obj):
     return isinstance(obj, (str, dict, io.IOBase))
 
 
+def _islice_index(sequence, *args):
+    r"""Yield indices for a given slice applied to a specific sequence.
+
+    Args:
+        sequence (Sequence[Any]):
+            Any sequence-like object.
+        \*args:
+            Either a slice object or the start, stop, and step of the slice
+            explicitly given.
+
+    Yields:
+        int:
+            The indices corresponding to the slice applied to the sequence.
+    """
+    if len(args) == 1 and isinstance(args[0], slice):
+        slice_ = args[0]
+        args = [slice_.start, slice_.stop, slice_.step]
+    yield from itertools.islice(list(range(len(sequence))), *args)
```

Comment:
The `list` call might be unnecessary, based on my quick tests. `range` objects can be sliced directly to produce another `range` object.

However, you must ensure that all values are positive here, you can't allow negative indexing in `islice` like `stop=-1`.

Separately, do you want to handle the case of a single positional argument `stop`? That doesn't seem to be handled (or documented) right now.

I think it might need to look like this?
```suggestion
    if len(args) == 1:
        args[0] = _handle_index(args[0])
    else:
        args = [_handle_index(args[0]), _handle_index(args[1]), args[2]]
    yield from itertools.islice(list(range(len(sequence))), *args)
```

Reply:
Hmm, I didn't realize that islice didn't handle the full slice specification... I might just revert the change then, because that means, I will have to effectively code the full logic anyways to handle all slices. Also, `stop` is handled since `islice(seq, stop)` works and we just pass the args along unless the passed object was a slice object.